### PR TITLE
chore: add merge gate keeper

### PR DESCRIPTION
**Pull request by alon-dotan-starkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/253)

### CODE REVIEW

**Review by dorimedini-starkware — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/253#-:-O3CkMrF0h5w8jq45ZvU:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @alon-dotan-starkware)

___
*[`.github/workflows/main.yml` line 169 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/253#-O3CkJHW1Pprcrb2LOF1:-O3CkJHW1Pprcrb2LOF2:b-vdqqxu) ([raw file](https://github.com/starkware-libs/sequencer/blob/b13585495d099b42fa5382e5ab0413ac629b85df/.github/workflows/main.yml#L169)):*
> ```YAML
>       - run: cargo check --workspace -r --all-features
> 
>   merge-gatekeeper:
> ```

cool!
GH settings updated to block on this?



_Code quote:_
```YAML
merge-gatekeeper:
```

**Review by alon-dotan-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/253)* status: all files reviewed, 1 unresolved discussion (waiting on @dorimedini-starkware)

___
*[`.github/workflows/main.yml` line 169 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/253#-O3CkJHW1Pprcrb2LOF1:-O3ClFvp5G7u5OekiqL5:bppk3bd) ([raw file](https://github.com/starkware-libs/sequencer/blob/b13585495d099b42fa5382e5ab0413ac629b85df/.github/workflows/main.yml#L169)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

cool!
GH settings updated to block on this?
</blockquote></details>

only after merge...

**Pull request merged**
